src/main/java/organice/logic/commands/DoneCommand.java

```diff
@@ -71,26 +78,58 @@ public boolean isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model mo
         }
         if (model.hasPatient(patientNric) && model.hasDonor(donorNric)
                 && match(donor, patient)) {
-            //todo check if tasklist isempty
-            return true;
+            return donor.getStatus().isProcessing() && patient.getStatus().isProcessing();
         } else {
             return false;
         }
+    }
 
+    /**
+     * A boolean to determine the value of the String, result
+     * @param result a String from the user input
+     * @return true if the result shows a pass, false if the result shows a fail.
+     */
+    public boolean isPass(String result) {
+        if (result.toLowerCase().equals("pass")) {
+            return true;
+        } else if (result.toLowerCase().equals("fail")) {
+            return false;
+        } else {
+            return false;
+        }
     }
 
+    /**
+     * Set the status of the patient and donor to done if the result of the cross-matching is pass.
+     * Set the status of the patient and donor to not processing if the result of the cross-matching
+     * fail. Then it will be taken into consideration when MatchCommand is called.
+     * @param model {@code Model} which the command should operate on.
+     * @return CommandResult object.
+     */
+
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
         try {
-            if (isValidDonorPatientPair(firstNric, secondNric, model)) {
+            if (isValidDonorPatientPair(firstNric, secondNric, model) && isPass(result)) {
+
                 model.getFilteredPersonList();
 
                 donor.setStatus(statusDone);
```

Comment:
If you are going to set the status as 'done' then delete it, why not delete the persons directly?

Reply:
explained in the javadoc, but basically its for future development